### ISSUE TITLE
cityscapes mask mAP in coco protocol maybe higher than in standard cityscapes protocol?

### ISSUE THREAD

**Issue by nemonameless** [open-mmlab/mmdetection#1414]
I think your mAP in coco protocol is not in conformity with the protocol of cityscapes evaluation https://github.com/mcordts/cityscapesScripts .Although your mAP is higher than the original MASK RCNN paper, it may be pseudomorph. I have trained cityscapes(only fine) using your code and even get 43.0 mask mAP in val-set in coco protocol, but it is only 30.6 when evaluate the val-set with cityscapesScripts,and only 27.3 on the leaderboard(test-set) of cityscapes.

And could you provide your instancesonly_filtered_gtFine_val.json and instancesonly_filtered_gtFine_train.json? Thanks.

I follow the maskrcnn-benchmark and get jsons, but when testing your trained model(https://open-mmlab.s3.ap-northeast-2.amazonaws.com/mmdetection/models/cityscapes/mask_rcnn_r50_fpn_1x_city_20190727-9b3c56a5.pth), I onlt get only 23.9 bbox mAP and 19.9 mask mAP in coco protocol.

**Comment by michaelisc** [comment]
Thanks for pointing this out. The reliance on the COCO evaluation tools is indeed a weakness of the current implementation and I did not test the results with the original cityscapes evaluation tools before submitting the pull request. The values you report are around 2% worse than those achieved with the current pycocotools implementation, so it is not completely off. Still 2% is quite a margin and I will have to look into that. I'll also try to add the original cityscapes evaluation in the future but if you have working code to do this I would very much appreciate your contribution. 

A question: What type of model did you train to get 43.0 % mask AP and was it the same model, which achieves 30.6% on the cityscapes val set? Because that would be a huge difference and rather weird. Considering that our pretrained model works so badly there must be something else going wrong, potentially in the annotation conversion. I still have adding a custom annotation conversion on my todo list. Sharing the annotations publicly will however not be possible I guess, because of the access-only model under which cityscapes is distributed.

**Comment by nemonameless** [comment]
@michaelisc Thanks for your reply. I trained HTC-ResNeXt101 (Hybrid Task Cascade) with coco-pretrain and get 43.0  % mask AP in val-set  in COCO evaluation protocol.   But when I submit the test-set result using this model， it's only 27.3（https://www.cityscapes-dataset.com/anonymous-results/?id=e781f9baaf8a69c045bfa15dd366a5a64d7d1c53b7c2a10606080a2ce09279fc）, which should be the standard cityscapes evaluation protocol.  

I sincerely hope that you can share your instancesonly_filtered_gtFine_val.json, and  mine can also email to you. If we keep the same val-set json file, the comparison on the validation set will be meaningful. Of course, it's also ok to submit test-set to see the real mask mAP, but it's time-consuming (only can submit once in 48 hours).
I  follow this(https://github.com/facebookresearch/maskrcnn-benchmark/pull/1088 )  but I am still not sure whether it is right or not.  I also hope you can evaluate your val-set result (32.5 mask mAP) by this tool (https://github.com/mcordts/cityscapesScripts). 

I've sent you an email before propose this issue. We can also communicate by email afterwards. Thanks very much.

**Comment by johnlanbor** [comment]
@nemonameless Hi, can you tell me how you fintune your model, I just change cls and seg weights to coco, but only got 39.6AP, Thanks

**Comment by nemonameless** [comment]
@johnlanbor I also  just change cls and seg weights. The coco pretrain model is HTC(mAP 50.7, 43.9). Can you share your jsons ( instancesonly_filtered_gtFine_val.json and instancesonly_filtered_gtFine_train.json) to me for further experiments?  nemonameless@qq.com, Thanks.

**Comment by johnlanbor** [comment]
@nemonameless sure, I will email you

**Comment by hoonkai** [comment]
@nemonameless Did you fine-tune the semantic branch as well?

**Comment by Ranchentx** [comment]
@nemonameless Hi, I generate the instancesonly_filtered_gtFine_val.json according to (facebookresearch/maskrcnn-benchmark#1088 ) , and using the model of mmdetection with AP 36 to test, but only 27 AP.  Can you share your jsons ( instancesonly_filtered_gtFine_val.json)? my email is 1297980835@qq.com. Thanks a lot.

**Comment by yhcao6** [comment]
Currently, we have reimplemented cityscapes. The results are evaluated exactly by the tool (https://github.com/mcordts/cityscapesScripts) you have mentioned. Please check it again. Feel free to reopen this issue.
